### PR TITLE
#277 Fix incorrect recipient

### PR DESCRIPTION
**Pull request by oubchid**

### CODE REVIEW

File: src/esn.inbox.libs/app/services/send-email/email-sending.service.js [hunk 1]
```diff
@@ -193,13 +193,17 @@ angular.module('esn.inbox.libs')
       };
     }
 
-    function getReplyRecipients(email) {
+    function getReplyRecipients(email, sender) {
+      function notMe(item) {
+        return item.email !== getEmailAddress(sender);
+      }
```
Comment (rezk2ll) [inline src/esn.inbox.libs/app/services/send-email/email-sending.service.js:199]:
```suggestion
      const notMe = ({ email }) => email !== getEmailAddress(sender);
```

**Review by rouazana — COMMENTED** [review]
why don't you add any test case about multiple recipient (several to, several cc, several bcc, mixed)?

File: src/esn.inbox.libs/app/services/send-email/email-sending.service.js [hunk 2]
```diff
@@ -193,13 +193,17 @@ angular.module('esn.inbox.libs')
...
+
       if (!email) {
         return;
       }
 
       return {
-        to: getReplyToRecipients(email),
+        to: _(email.to || []).concat(getReplyToRecipients(email)).uniq('email').value().filter(notMe),
```
Comment (rezk2ll) [inline src/esn.inbox.libs/app/services/send-email/email-sending.service.js:206]:
```suggestion
        to: _.uniq([...getReplyToRecipients(email), email.to], 'email').filter(notMe),
```

**Comment by oubchid** [comment]
> why don't you add any test case about multiple recipient (several to, several cc, several bcc, mixed)?

I will , but first I need the fix the bug then fix the existing tests

**Comment by oubchid** [comment]
@rouazana we don't need the cc and bcc. this PR should fix only `to` (one recipient or multiple)
usera send a message to userb 
1. we should have userb in `to` when usera reply
usera send a message to userb and userc
2. we should have userb and userc in `to` when usera reply.


should keep all bcc, cc if we reply sush as reply all ?
if yes This PR should also fix bcc, and cc ?

**Comment by rouazana** [comment]
> should keep all bcc, cc if we `reply` sush as `reply all` ?
> if yes This PR should also fix bcc, and cc ?

If you are asking, this means that these tests are missing :)
In my opinion, yes reply all should be also handled, but it could be in an other ticket

**Review by rezk2ll — APPROVED** [review]

**Review by alagane — DISMISSED** [review]

**Comment by fabienmoyon** [comment]
typo in the commit and the PR title

**Review by fabienmoyon — CHANGES_REQUESTED** [review]
typo in the commit and the PR title

**Review by alagane — APPROVED** [review]

**Review by fabienmoyon — APPROVED** [review]

**Pull request merged**